File: baas/src/main/scala/com/ing/baker/recipe/commonserialize/Interaction.scala
package com.ing.baker.recipe.commonserialize

import com.ing.baker.recipe.common

case class Interaction(override val name: String,
                       override val inputIngredients: Seq[Ingredient],
                       override val output: Seq[common.Event])
  extends common.Interaction {

  def this(interaction: common.Interaction) =
    this(
      interaction.name,
      interaction.inputIngredients.map(i => new Ingredient(i)),
      interaction.output)

  override def equals(obj: Any): Boolean = {
    if (!obj.isInstanceOf[common.Interaction])
      return false
    val other: common.Interaction = obj.asInstanceOf[common.Interaction]
    this.name == other.name &&
      this.inputIngredients == other.inputIngredients &&
      this.output == other.output
  }
}

File: core/baker-compiler/src/main/kotlin/com/ing/baker/recipe/Interaction.kt
package com.ing.baker.recipe

import com.ing.baker.recipe.common.InteractionFailureStrategy
import com.ing.baker.types.Converters
import com.ing.baker.types.Value

data class Interaction(
    override val name: String,
    val oldName: String? = null,
    override val inputIngredients: List<Ingredient> = emptyList(),
    override val output: List<Event> = emptyList(),
    override val requiredEvents: Set<String> = emptySet(),
    override val requiredOneOfEvents: Set<Set<String>> = emptySet(),
    override val predefinedIngredients: Map<String, Value> = emptyMap(),
    override val overriddenIngredientNames: Map<String, String> = emptyMap(),
    override val overriddenOutputIngredientName: String? = null,
    override val eventOutputTransformers: Map<Event, EventOutputTransformer> = emptyMap(),
    override val maximumInteractionCount: Int? = null,
    override val failureStrategy: InteractionFailureStrategy? = null,
    override val isReprovider: Boolean = false,
) : InteractionDescriptor {
    override val originalName get() = oldName ?: name
}
